Data.Rmd

```diff
@@ -0,0 +1,329 @@
+This means that they must be documented.
+Documenting data is like documenting a function with a few minor differences.
+Instead of documenting the data directly, you document the name of the dataset and save it in `R/`.
+For example, the roxygen2 block used to document the `who` data in tidyr is saved in `R/data.R` and looks something like this:
+
+```{r, eval = FALSE}
+#' World Health Organization TB data
+#'
+#' A subset of data from the World Health Organization Global Tuberculosis
+#' Report ...
+#'
+#' @format ## `who`
+#' A data frame with 7,240 rows and 60 columns:
+#' \describe{
+#'   \item{country}{Country name}
+#'   \item{iso2, iso3}{2 & 3 letter ISO country codes}
+#'   \item{year}{Year}
+#'   ...
+#' }
+#' @source <https://www.who.int/teams/global-tuberculosis-programme/data>
+"who"
+```
+
+There are two roxygen tags that are especially important for documenting datasets:
+
+-   `@format` gives an overview of the dataset.
+    For data frames, you should include a definition list that describes each variable.
+    It's usually a good idea to describe variables' units here.
+
+-   `@source` provides details of where you got the data, often a URL.
+
+Never `@export` a data set.
+
+### Non-ASCII characters in data
+
+The R objects you store in `data/*.rda` often contain strings, with the most common example being character columns in a data frame.
+If you can constrain these strings to only use ASCII characters, it certainly makes things simpler.
+But of course, there are plenty of legitimate reasons why package data might include non-ASCII characters.
+
+In that case, we recommend that you embrace the [UTF-8 Everywhere manifesto](http://utf8everywhere.org) and use the UTF-8 encoding.
+The `DESCRIPTION` file placed by `usethis::create_package()` always includes `Encoding: UTF-8`, so by default a devtools-produced package already advertises that it will use UTF-8.
+
+Making sure that the strings embedded in your package data have the intended encoding is something you accomplish in your data preparation code, i.e. in the R scripts below `data-raw/`.
+You can use `Encoding()` to learn the current encoding of the elements in a character vector and functions such as `enc2utf8()` or `iconv()` to convert between encodings.
+
+::: callout-warning
+## Submitting to CRAN
+
+If you have UTF-8-encoded strings in your package data, you may see this from `R CMD check`:
+
+    -   checking contents of 'data' directory ... OK
+    -   checking data for non-ASCII characters ... NOTE
+        Note: found 352 marked UTF-8 strings
+
+This `NOTE` is truly informational, i.e. it is a `NOTE` that you do not need to address or attempt to fix.
+As long as you actually intend to have UTF-8 strings in your package data, all is well.
+
+Ironically, this `NOTE` is actually suppressed by `R CMD check --as-cran`, i.e. by including the `--as-cran` flag, despite the fact that this note does appear in the check results once a package is on CRAN.
```

Comment:
Even more ironically, we have recently learned that `--as-cran` is not used by CRAN.

Reply:
I tweaked the callout to state this more clearly.